Week3/App.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/* global Util, Repository, Contributor */
+
+class App {
+  constructor(url) {
+    this.initialize(url);
+  }
+
+  /**
+   * Initialization
+   * @param {string} url The GitHub URL for obtaining the organization's repositories.
+   */
+  async initialize(url) {
+    // Add code here to initialize your app
+    // 1. Create the fixed HTML elements of your page
+    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
+
+    const root = document.getElementById('root');
+    const header = Util.createAndAppend("header", root, { id: "mainCon" });
+    const article = Util.createAndAppend("article", root, { id: "container" });
+    Util.createAndAppend("p", header, { text: "HYF  Repositories", id: "HYF_Repositories" });
+    const select = Util.createAndAppend("select", header);
+
+    try {
+
+      const repos = await Util.fetchJSON(url);
+      this.repos = repos.sort((a, b) => a.name.localeCompare(b.name)).map(repo => new Repository(repo));
```

Comment:
It is customary to split a method chain over multiple lines if the line becomes long, e.g.:

```js
this.repos = repos
  .sort((a, b) => a.name.localeCompare(b.name))
  .map(repo => new Repository(repo));
```